java/code/src/com/suse/manager/webui/services/pillar/MinionPillarFileManager.java

```diff
@@ -54,7 +54,12 @@ public MinionPillarFileManager(MinionPillarGenerator minionPillarGeneratorIn) {
      */
     public void generatePillarFile(MinionServer minion) {
         SaltPillar pillar = this.minionPillarGenerator.generatePillarData(minion);
-        this.saveFileToDisk(pillar, this.minionPillarGenerator.getFilename(minion.getMinionId()));
+        if (!pillar.isEmpty()) {
```

Comment:
Looks good. I would change it the generatePillarData function to return an Optional<SaltPillar>, and then put 

this.minionPillarGenerator.generatePillarData(minion);
        Optional<SaltPillar> o = Optional.empty();
        o.ifPresent(() -> this.saveFileToDisk(pillar, this.minionPillarGenerator.getFilename(minion.getMinionId())));